cmd/new.go

```diff
@@ -112,12 +116,165 @@ func scaffoldRule(params newRuleCommandParams) error {
 	}
 
 	if params.type_ == "builtin" {
+		if err := addToDataYAML(params); err != nil {
+			return err
+		}
+
+		if err := addRuleToREADME(params); err != nil {
+			return err
+		}
+
 		return scaffoldBuiltinRule(params)
 	}
 
 	return fmt.Errorf("unsupported type %v", params.type_)
 }
 
+func addToDataYAML(params newRuleCommandParams) error {
+	// Read the YAML content from the file
+	yamlContent, err := os.ReadFile("bundle/regal/config/provided/data.yaml")
+	if err != nil {
+		return err
+	}
+
+	var existingConfig config.Config
+
+	// Unmarshal the YAML content into a map
+	if err := yaml.Unmarshal(yamlContent, &existingConfig); err != nil {
+		return err
+	}
+
+	// Add the new entry to the Rules map within the Config struct
+	if existingConfig.Rules == nil {
```

Comment:
This should never be nil, as the provided configuration will always have rules, right?